File: tiendapome.frontend/src/app/models/categoria.ts
import { Subcategoria } from './subcategoria';
import { Tipo } from './tipo';

export class Categoria {
    Id: number;
    Descripcion?: string;
    Vigente: boolean;
    Tipo: Tipo;
    Carpeta: string;
    Foto: string;
    Subcategorias: Array<Subcategoria>;
    CantidadProductos: number;
    Orden: number;
    Visible: boolean;
    IdTipo: number;

    constructor(
        _id: number
    ) { 
        this.Id = _id;
    }
}

File: tiendapome.backend/tiendapome.Entidades/Categoria.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace tiendapome.Entidades
{
    [JsonObject]
    public class Categoria : EntidadBaseParametrica
    {
        [JsonIgnore]
        public virtual Tipo Tipo { get; set; }

        [JsonProperty("Carpeta")]
        public virtual string Carpeta { get; set; }

        [JsonProperty("Foto")]
        public virtual string Foto { get; set; }

        [JsonProperty("Subcategorias")]
        public virtual IList<Subcategoria> Subcategorias { get; set; }

        [JsonProperty("CantidadProductos")]
        public virtual int CantidadProductos { get; set; }

        [JsonProperty("Orden")]
        public virtual int Orden { get; set; }

        [JsonProperty("Visible")]
        public virtual bool Visible { get; set; }

        [JsonProperty("IdTipo")]
        public virtual int IdTipo { get; set; }

        public Categoria() { }
    }
}
